crates/nu-command/src/core_commands/do_.rs

```diff
@@ -106,7 +109,8 @@ impl Command for Do {
             block,
             input,
             call.redirect_stdout,
-            capture_errors || ignore_shell_errors || ignore_program_errors,
+            // TODO remove capture_errors once redirection is fixed
+            call.redirect_stderr || capture_errors,
```

Comment:
I think in nushell `call.redirect_stderr` will always be `false`.

I'd suggest change the line to

`capture_errors || !ignore_all_errors`, so `do -i` will not redirect stdout, and standard `do` without `-i` will capture stderr and have a chance to collect by `complete`.